osu.Game/Screens/Play/HUD/DrawableGameplayLeaderboardScore.cs

```diff
@@ -107,324 +114,290 @@ public DrawableGameplayLeaderboardScore(GameplayLeaderboardScore score)
-                                        },
-                                    },
-                                },
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Font = OsuFont.Style.Caption1.With(weight: FontWeight.SemiBold),
+                                    Shear = -OsuGame.SHEAR,
+                                }
                             }
                         },
-                        gridContainer = new GridContainer
+                    },
+                    // this is placed here between the left and right layer for layout purposes,
+                    // but it's proxied below to render in front of them.
+                    avatarLayer = new Container
+                    {
+                        Size = new Vector2(avatar_size),
+                        // precise padding so the avatar's top and bottom sides land as close to the panel borders as possible.
+                        Padding = new MarginPadding(1.3f),
+                        // negative left margin to place the avatar's center directly at the edge of the left layer.
+                        Margin = new MarginPadding { Left = -avatar_size / 2 },
+                        Child = new Container
                         {
-                            RelativeSizeAxes = Axes.Y,
-                            Width = compact_width, // will be updated by expanded state.
-                            Anchor = Anchor.TopRight,
-                            Origin = Anchor.TopRight,
-                            ColumnDimensions = new[]
+                            RelativeSizeAxes = Axes.Both,
+                            CornerRadius = corner_radius,
+                            Masking = true,
+                            Child = new ScoreAvatar(User)
                             {
-                                new Dimension(GridSizeMode.Absolute, rank_text_width),
-                                new Dimension(),
-                                new Dimension(GridSizeMode.AutoSize),
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                RelativeSizeAxes = Axes.Both,
+                                Shear = -OsuGame.SHEAR,
+                                // extra scaling to cover the entire sheared area.
+                                Scale = new Vector2(1.1f),
                             },
-                            Content = new[]
+                        },
+                    },
+                    rightLayer = new Container
+                    {
+                        Width = right_panel_width,
+                        RelativeSizeAxes = Axes.Y,
+                        // negative left margin to make the X position of the right layer directly at the avatar center (rendered behind it).
```

Comment:
if *any*of this breaks ever, you know whoever needs to fix this code is probably going to rewrite it.